lib/membrane_overlay_filter.ex

```diff
@@ -0,0 +1,114 @@
+defmodule Membrane.OverlayFilter do
+  @moduledoc """
+  Applies image or text overlay to video.
+
+  Based on `Image`.
+  """
+  use Membrane.Filter
+
+  alias Membrane.RawVideo
+
+  def_input_pad :input, accepted_format: %RawVideo{pixel_format: :I420}
```

Comment:
Since we rely that heavy on the fact, that the YUV pixels are written in the `planar` format, perhaps it would be good to add a `pixels_order: :planar | :semiplanar | :interleaved` field to the `RawVideo` format?

Reply:
You are right, I420 stands for fully planar ;) take a look here: https://wiki.videolan.org/YUV#Planar_formats